mediadevices.go

```diff
@@ -142,18 +145,55 @@ func selectBestDriver(filter driver.FilterFn, constraints MediaTrackConstraints)
 }
 
 func (m *mediaDevices) selectAudio(constraints MediaTrackConstraints) (Tracker, error) {
-	d, c, err := selectBestDriver(driver.FilterAudioRecorder(), constraints)
+	filter := driver.FilterAudioRecorder()
+	if constraints.DeviceID != "" {
+		filter = func(d driver.Driver) bool {
+			return driver.FilterAudioRecorder()(d) && driver.FilterID(constraints.DeviceID)(d)
```

Comment:
I think it's more readable if it's written like the following,

```
audioFilter := driver.FilterAudioRecorder() // or even just filter
idFilter := driver.FilterID(constraints.DeviceID)
filter = func(d driver.Driver) bool {
	return audioFilter(d) && idFilter(d)
```

But, honestly, I don't have a hard feeling on this, just giving an opinion ^^. Feel free to ignore it.